client/src/main/java/io/pravega/client/stream/SegmentReader.java

```diff
@@ -41,4 +41,12 @@ public interface SegmentReader<T> extends AutoCloseable {
      * @return a future.
      */
     CompletableFuture<Void> isAvailable();
+
+    /**
+     * Gets the current snapshot of segment reader. It will provide information about segment reader's position,
+     * segment and if reader reads all the events in segment of not.
```

Comment:
statement can be slightly modified if that make sense - `segment itself and whether the reader has processed all events within the segment`.

Reply:
Done.